File: BackendProject/InventoryManagementAPI/Models/Product.cs
using System.Text.Json.Serialization;

namespace InventoryManagementAPI.Models
{
    public class Product
    {
        public int ProductId { get; set; }
        public string SKU { get; set; } = string.Empty; // Stock Keeping Unit
        public string ProductName { get; set; } = string.Empty;
        public string? Description { get; set; }
        public decimal UnitPrice { get; set; }
        public bool IsDeleted { get; set; }
        public int CategoryId { get; set; }

        // Navigation property
        public Category? Category { get; set; }
        [JsonIgnore]
        public ICollection<InventoryProduct>? InventoryProducts { get; set; }
    }
}

File: FrontendProject/inventory-app/src/app/models/product.ts
export interface Product {
  productId: number;
  sku: string;
  productName: string;
  description: string;
  unitPrice: number;
  categoryId: number;
  categoryName: string; 
  isDeleted: boolean;
}


export interface AddProductRequest {
  sku: string;
  productName: string;
  description: string;
  unitPrice: number;
  categoryId: number;
}


export interface UpdateProductRequest {
  productId: number;
  sku: string;
  productName: string;
  description: string;
  unitPrice: number;
  categoryId: number;
}